packages/monaco/src/browser/monaco-quick-command-service.ts

```diff
@@ -0,0 +1,132 @@
+    }
+
+    reset(): void {
+        const { recent, other } = this.getCommands();
+        this.recentItems = [];
+        this.otherItems = [];
+        this.recentItems.push(...recent.map(command => this.toItem(command)));
+        this.otherItems.push(...other.map(command => this.toItem(command)));
+    }
+
+    getPicks(filter: string, token: monaco.CancellationToken): monaco.quickInput.Picks<monaco.quickInput.IAnythingQuickPickItem> {
+        const items: Array<monaco.quickInput.IAnythingQuickPickItem> = [];
+        if (this.recentItems.length === 0 && this.otherItems.length === 0) {
+            this.reset();
+        }
+        const recentItems = filterItems(this.recentItems.slice(), filter);
+        const otherItems = filterItems(this.otherItems.slice(), filter);
+
+        if (recentItems.length > 0) {
+            items.push({ type: 'separator', label: 'recently used' }, ...recentItems);
+        }
+
+        if (otherItems.length > 0) {
+            items.push({ type: 'separator', label: 'other commands' }, ...otherItems);
+        }
+        return items;
+    }
+
+    private toItem(command: Command): monaco.quickInput.IAnythingQuickPickItem {
+        const label = (command.category) ? `${command.category}: ` + command.label! : command.label!;
+        const iconClasses = this.getItemIconClasses(command);
+        const activeElement = window.document.activeElement as HTMLElement;
+
+        return {
+            label,
+            iconClasses,
+            alwaysShow: !!this.commandRegistry.getActiveHandler(command.id),
+            keybinding: this.getKeybinding(command),
+            accept: () => {
+                activeElement.focus({ preventScroll: true });
+                this.commandRegistry.executeCommand(command.id);
+                this.commandRegistry.addRecentCommand(command);
+            }
+        };
+    }
+
+    private getKeybinding(command: Command): monaco.keybindings.ResolvedKeybinding | undefined {
+        const keybindings = this.keybindingRegistry.getKeybindingsForCommand(command.id);
+        if (!keybindings || keybindings.length === 0) {
+            return undefined;
+        }
+
+        let keySequence: KeySequence;
+        try {
+            keySequence = this.keybindingRegistry.resolveKeybinding(keybindings[0]);
+        } catch (error) {
+            return undefined;
+        }
+        return new MonacoResolvedKeybinding(keySequence, this.keybindingRegistry);
+    }
+
+    private getItemIconClasses(command: Command): string[] | undefined {
+        const toggledHandler = this.commandRegistry.getToggledHandler(command.id);
+        if (toggledHandler) {
+            return ['fa fa-check'];
+        }
+        return undefined;
+    }
+}
+
+export class CommandsQuickAccessProvider extends monaco.quickInput.PickerQuickAccessProvider<monaco.quickInput.IQuickPickItem> {
+    static PREFIX = '>';
+    static dataService: monaco.quickInput.IQuickAccessDataService;
```

Comment:
Why would you use a static variable here instead of dependency injection?

Reply:
`CommandsQuickAccessProvider` is provider used by monaco and so I didn't succeed in injecting Theia classes into Monaco class. Adding an inject to it was not successful. It is basically handled and initialized by monaco when doing registerQuickAccessProvider. 

Please advise if you think it is possible.